### PR TITLE
always set cgroup driver for containerd and kubelet to systemd for new nodes. Don't touch existing nodes.

### LINKED ISSUE #98: cgroup driver configured inconsistently for containerd and kubelet
**How to categorize this issue?**

/area os
/kind bug
/os garden-linux

**What happened**:

The scripts introduced in PR #82 and PR #85 are supposed to configure containerd's cgroup driver according to the cgroup version of the underlying host and to make sure that the kubelet is configured with the corresponding cgroup driver. Both reconfigurations happen by running a script in the respectives component's `ExecStartPre=` before it comes up.

Since these changes were released to Gardener landscapes, they proved to be reliable for most of the time. However, we came across at least one instance of a new node with Garden Linux **934.7** and hence with cgroup v2 being scheduled for an existing worker pool which failed to join the cluster due to an inconsistent cgroup driver configuration: containerd ran with the `systemd` cgroup driver but the kubelet still ran with the `cgroupfs` cgroup driver.

From the logs of the internal support ticket:

```
2023-05-10 06:50:55 | {"_entry":"Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: expected cgroupsPath to be of format \"slice:prefix:name\" for systemd cgroups, got \"/kubepods/burstable/pod8d7a925b-ba2f-4fcd-9eb7-6a933858f3df/d6f8417b97f66a8e2fd5b7c23cf01222a1b7b476c29c12b1d557c219061e96db\" instead: unknown","count":1,"firstTimestamp":"2023-05-10T06:50:55Z","lastTimestamp":"2023-05-10T06:50:55Z","namespace":"kube-system","object":"Pod/egress-filter-applier-sx4d4","origin":"shoot","reason":"FailedCreatePodSandBox","source":"kubelet","sourceHost":"shoot--hyperspace--prod-int-worker-gqla1-z1-67c55-ntm74","type":"Warning"}
2023-05-10 06:50:55 | {"_entry":"Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: expected cgroupsPath to be of format \"slice:prefix:name\" for systemd cgroups, got \"/kubepods/burstable/pod2eed3a6f-3e07-42f9-9219-ba72f7f66c47/d85c081a0e7821d8e5a853ee48e97bc0bf2cebb8050b4ff9ac0a3ca9e10ae50b\" instead: unknown","count":1,"firstTimestamp":"2023-05-10T06:50:55Z","lastTimestamp":"2023-05-10T06:50:55Z","namespace":"kube-system","object":"Pod/apiserver-proxy-6fdfb","origin":"shoot","reason":"FailedCreatePodSandBox","source":"kubelet","sourceHost":"shoot--hyperspace--prod-int-worker-gqla1-z1-67c55-ntm74","type":"Warning"}Show context
2023-05-10 06:50:55 | {"_entry":"Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: expected cgroupsPath to be of format \"slice:prefix:name\" for systemd cgroups, got \"/kubepods/burstable/pod0afb0879-f205-4b66-bc0b-28ff62c898b2/167ccf567fa2cd2a438b18b63b158c0bb21c859559e93924f2f2d1234c35d17b\" instead: unknown","count":1,"firstTimestamp":"2023-05-10T06:50:55Z","lastTimestamp":"2023-05-10T06:50:55Z","namespace":"kube-system","object":"Pod/node-local-dns-hhns4","origin":"shoot","reason":"FailedCreatePodSandBox","source":"kubelet","sourceHost":"shoot--hyperspace--prod-int-worker-gqla1-z1-67c55-ntm74","type":"Warning"}
2023-05-10 06:50:55 | {"_entry":"Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: expected cgroupsPath to be of format \"slice:prefix:name\" for systemd cgroups, got \"/kubepods/burstable/poda592f80e-1070-482b-94db-1b070e836f7a/930ac369e7035000c8810e6c72b720ff73485834269738d930d18ef96b489874\" instead: unknown","count":1,"firstTimestamp":"2023-05-10T06:50:55Z","lastTimestamp":"2023-05-10T06:50:55Z","namespace":"kube-system","object":"Pod/csi-driver-node-gm8p4","origin":"shoot","reason":"FailedCreatePodSandBox","source":"kubelet","sourceHost":"shoot--hyperspace--prod-int-worker-gqla1-z1-67c55-ntm74","type":"Warning"}
```

These log message are a strong pointer to the kubelet creating a classic cgroupfs hierarchy which containerd then has not clue about what to do with.

**What you expected to happen**:

Containerd and kubelet are always running with the same cgroup driver and log message like the ones given above do not occur.

**How to reproduce it (as minimally and precisely as possible)**:

Steps to reproduce need to be investigated. It seems that this is caused by a timing issue or a race condition.

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**How to categorize this PR?**

/area os
/kind bug
/os garden-linux

**What this PR does / why we need it**:

Always sets cgroup driver for containerd and kubelet to systemd for new nodes

Why required:
- fixes a bug in the extension that could lead under certain race conditions to the kubelet and containerd using different cgroup managers (causes node to not being able to join)
- simplifies the startup-logic because dependencies between containerd and the kubelet configuration are removed

**Which issue(s) this PR fixes**:
Fixes #98 

**Special notes for your reviewer**:

**Release note**:

```breaking operator
hardcoded cgroup driver for containerd and kubelet to systemd for ALL new nodes. Requires the Gardener installation to only have Gardenlinux versions with cgroups-v2 only. This includes GL 934 and up.
```

### CODE REVIEW

**Comment by danielfoehrKn** [comment]
I am not 100% done with my testing here yet. Not seeing problems so far, but wanted to be extra thorough. Ready to be reviewed.

**Comment by danielfoehrKn** [comment]
Tested the PR now for existing and new clusters and I don't see problems. So it's ready for review and testing.

File: pkg/controller/operatingsystemconfig/actuator_test.go [hunk 1]
```diff
@@ -348,35 +287,29 @@ set -Eeuo pipefail
...
-if check_running_containerd_tasks; then
-    configure_containerd "$(check_current_cgroup)"
-
-    # in rare cases it could be that the kubelet.service was already running when
-    # containerd got reconfigured so we restart it to force its ExecStartPre
-    if systemctl is-active kubelet.service; then
-        echo "triggering kubelet restart..."
-        systemctl restart --no-block kubelet.service
-    fi
+# do not change the kubelet's configuration on an existing system with running containers
```
Comment (MrBatschner) [inline pkg/controller/operatingsystemconfig/actuator_test.go:303]:
```suggestion
# do not change the containerd's configuration on an existing system with running containers
```

**Review by MrBatschner — COMMENTED** [review]
Except for two (really overly pedantic) nits, it looks good to me.

**Comment by danielfoehrKn** [comment]
Thanks for the review. Updated the PR

**Review by MrBatschner — APPROVED** [review]
/lgtm

**Pull request merged**